cheroot/test/test_server.py

```diff
@@ -236,3 +237,65 @@ def test_peercreds_unix_sock_with_lookup(peercreds_enabled_server):
         peercreds_text_resp = requests.get(unix_base_uri + PEERCRED_TEXTS_URI)
         peercreds_text_resp.raise_for_status()
         assert peercreds_text_resp.text == expected_textcreds
+
+
+@pytest.mark.skipif(
+    IS_WINDOWS,
+    reason=(
+        'This regression test is for a Linux bug, and the resource module '
+        'is not available on Windows.'
+    ),
+)
+def test_high_number_of_file_descriptors():
+    """Test the server does not crash with a high file-descriptor value.
+
+    This test should cause a server crash, as the server will try to
+    select() a file-descriptor higher than 1024.
+    """
+    # Imported here, as it is not available on Windows
+    import resource
```

Comment:
Import statements must be on top. The conditionally available modules can be ignored with try/except idiom.

But there's another way. pytest has https://docs.pytest.org/en/stable/reference.html#pytest-importorskip helper and seems to be perfect for this use-case:
```suggestion
    resource = pytest.importorskip(
        'resource',
        reason='The "resource" module is Unix-specific',
    )
```